### PR TITLE
Replace usage of Bullet Physics with Rapier Physics

### PR DESCRIPTION
**Pull request by dgavedissian**
http://rapier.rs is a pure Rust physics library.

A side effect of the Rust port is that the physics code is still implemented in C++ with Bullet Physics. This ends up with a weird hack where we have to compile and link C++ code during the Rust build process. Moving to Rapier allows us to remove the last remaining part of C++ code, making compilation much faster. Rapier is also better performant, and is a prerequisite for us to implement true large world support.

This PR utilises the 64-bit version of Rapier, which means all physics calculations are done with 64-bit floats now. Once we've implemented relative rendering, and we represent positions in game logic in 64-bit, that should be enough for us to support large worlds properly.

### CODE REVIEW

File: engine/lib/phx/src/physics/physics.rs [hunk 1]
```diff
@@ -1,175 +1,525 @@
...
+                rp::QueryFilter::default(),
+            )
+            .is_some()
+    }
 }
 
-#[no_mangle]
-pub unsafe extern "C" fn Physics_DrawBoundingBoxesWorld(this: &mut Physics) {
-    _cppPhysics_DrawBoundingBoxesWorld(this)
+/// Convert HSL color to RGB
```
Comment (HaronK) [inline engine/lib/phx/src/physics/physics.rs:462]: Maybe it's better to move this to Color struct as you have suggested before?

**Review by HaronK — APPROVED** [review]
Overall looks good for me as a first iteration. Just a few small comments.

**Review by IllustrisJack — APPROVED** [review]

**Comment by HaronK** [comment]
Btw, would it be possible to switch to stable Rust with this PR?

**Comment by dgavedissian** [comment]
> Btw, would it be possible to switch to a stable Rust with this PR?

Yep I'll try and do this now.

File: engine/lib/phx/src/physics/physics.rs (hunk 1 above)
Comment (dgavedissian) [reply engine/lib/phx/src/physics/physics.rs:461]: I'd rather leave this for https://github.com/Limit-Theory-Redux/ltheory/pull/212 - as that's the PR that introduces the colour class.

File: rust-toolchain.toml [hunk 2]
```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "nightly-2023-09-28"
-components = ["rustfmt-preview", "rustc-dev", "rust-src", "miri"]
+channel = "stable"
+components = ["rustfmt"]
```
Comment (HaronK) [inline rust-toolchain.toml:3]: We still need `rust-src` for documentation in vscode

Comment (dgavedissian) [reply rust-toolchain.toml:3]: Fixed

File: engine/lib/phx/src/physics/physics.rs (hunk 1 above)
Comment (HaronK) [reply engine/lib/phx/src/physics/physics.rs:461]: Ah, yeah. I forgot color is not merged yet.

**Pull request merged**